### ISSUE TITLE
[a11y]: HeaderPanel requires an aria-label but Accessibility Checker complains if you have one.

### ISSUE THREAD

**Issue by kglickman** [carbon-design-system/carbon#13728]
### Package

@carbon/react

### Package version

1.23.0

### React version

16.14.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

The tool is complaining about the aria-label in the HeaderPanel.  If we remove it, the error goes away but we get an annoying error from checkPropTypes in the console saying it's required.
Warning: Failed prop type: HeaderPanel requires one of the following props: aria-label, aria-labelledby

![image](https://user-images.githubusercontent.com/30597411/236199630-6e45dd91-20df-45b1-9206-e59186dcc812.png)

 

### Reproduction/example

Reproducible in the story book

### Steps to reproduce

Look at the Accessibility section in the this url https://react.carbondesignsystem.com/?path=/story/components-ui-shell--header-base-w-actions-and-right-panel
https://react.carbondesignsystem.com/?path=/story/components-ui-shell--header-base-w-actions-and-right-panel

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by btepe** [comment]
We ran into the same issue. Using an `aria-label` without a role is not allowed. What is missing here is either an `aria-role` attribute on the `div` element used for the `HeaderPanel` or using a semantic HTML element instead of the generic `div`. I would suggest using an `aside` element.

@kglickman Until this is implemented in the library, we can set the `aria-role="complementary"`. This seems like the most appropriate role according to [MDN web docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/complementary_role).

**Comment by kglickman** [comment]
I don't feel like either an "aside" element or a role of "complementary" are appropriate for a header.  We would be silencing the Accessibility Checker but making the page more difficult to understand.

**Comment by btepe** [comment]
@kglickman Perhaps the role and element I suggested are not the best choice, which ones would you suggest using?

When you look at the [Carbon docs on the right header panel](https://carbondesignsystem.com/components/UI-shell-right-panel/usage/#general-guidance), it states that the right panel is an `optional panel that shows additional system level actions or content associated with a system icon in the header`. This means that it can contain a variety of interactive elements and plain information, so it isn't necessarily part of the navigation as are the other elements in the header. To me using a `nav` element doesn't seem like the right choice because of this.

**Comment by kglickman** [comment]
We removed the aria label and the Accessibility Checker doesn't report an error anymore.  This defect is about the prop being required, causing stacks in the console.

**Comment by tw15egan** [comment]
Needs to be backported to V10

**Comment by tw15egan** [comment]
Closed via #14517 , will be included in the next version of v10
